src/Model/Aggregate.php

```diff
@@ -0,0 +1,334 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\data\FieldSqlExpression;
+use atk4\data\Model;
+use atk4\data\Reference;
+use atk4\dsql\Query;
+
+/**
+ * Aggregate model allows you to query using "group by" clause on your existing model.
+ * It's quite simple to set up.
+ *
+ * $aggregate = new Aggregate($mymodel);
+ * $aggregate->groupBy(['first','last'], ['salary'=>'sum([])'];
+ *
+ * your resulting model will have 3 fields:
+ *  first, last, salary
+ *
+ * but when querying it will use the original model to calculate the query, then add grouping and aggregates.
+ *
+ * If you wish you can add more fields, which will be passed through:
+ * $aggregate->addField('middle');
+ *
+ * If this field exist in the original model it will be added and you'll get exception otherwise. Finally you are
+ * permitted to add expressions.
+ *
+ * The base model must not be Union model or another Aggregate model, however it's possible to use Aggregate model as nestedModel inside Union model.
+ * Union model implements identical grouping rule on its own.
+ *
+ * You can also pass seed (for example field type) when aggregating:
+ * $aggregate->groupBy(['first','last'], ['salary' => ['sum([])', 'type'=>'money']];
+ */
+class Aggregate extends Model
```

Comment:
For me, everything that except Model is very suboptimal - how can be model extended then in an user app (ok, with trait, but still)?